nmrfx-processor/src/main/java/org/nmrfx/processor/datasets/peaks/PeakFitter.java

```diff
@@ -882,7 +892,19 @@ public double doFit(int i0, int i1, int[] rows, boolean doFit, boolean linearFit
 
         int size = pt[0][1] - pt[0][0] + 1;
         Vec fitVec = new Vec(size);
-        theFile.readVectorFromDatasetFile(pt, fitDim, fitVec);
+
+        // The indices must be converted to raw indices to read the file, the real/complex indices are used later in
+        // the function so pt should not be modified.
+        int[][] ptRaw = new int[pt.length][2];
+        if (theFile.getComplex(pdim[0])) {
+            ptRaw[0][0] = pt[0][0] * 2;
+            ptRaw[0][1] = (pt[0][1] * 2) + 1;
+        } else {
+            ptRaw[0][0] = pt[0][0];
+            ptRaw[0][1] = pt[0][1];
+        }
+
+        theFile.readVectorFromDatasetFile(ptRaw, fitDim, fitVec);
```

Comment:
When getting the vector for fitting the peaks, the raw indices also need to be used to retrieve the vector, but also need to keep the real/complex indices because the fitting function adjusts those indices further down in the function